staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -213,6 +215,32 @@ func newAuthenticator(c *cache, config *api.ExecConfig, cluster *clientauthentic
 	return c.put(key, a), nil
 }
 
+func isInteractive(isTerminalFunc func(int) bool, config *api.ExecConfig, stdinUnavailable bool) (bool, error) {
+	var shouldBeInteractive bool
+	switch config.InteractiveMode {
+	case "":
+		// backwards compatible default
+		shouldBeInteractive = isTerminalFunc(int(os.Stdin.Fd()))
```

Comment:
Defaulting should allow you to drop this case.

Reply:
If folks are manually setting their `rest.Config.ExecConfig` today, then they will most likely run into a runtime error here upon upgrade with no changes to their code. Is this OK? Per some of our previous conversations about upgrading k8s.io Go code, it sounds like there is precedent for this.

Similar case for `clientcmd.Config` - folks that are setting their `clientcmd.Config` manually will most likely get a validation error on upgrade if they are using an `ExecConfig` and they don't set the `InteractiveMode` field.

See 54dd5407693621307d2dc61a9bcc2c3e832d472e for where I took a pass at defaulting.